penguin/web/sse_events.py

```
@@ -305,55 +316,47 @@ def event_handler(event_type: str, data: Any):
-    if len(buffer) > _MAX_REPLAY_EVENTS:
-        del buffer[: len(buffer) - _MAX_REPLAY_EVENTS]
+
+    def ledger_handler(event_type: str, data: Any) -> None:
+        if event_type != "opencode_event" or not isinstance(data, dict):
+            return
+        try:
+            record_opencode_event(core, data)
+        except Exception:
+            # Ledger failures must not interrupt live TUI streaming. The client can
+            # still receive the live frame; replay will surface a gap if needed.
+            pass
```

Ruff (0.15.20)

[warning] Use `contextlib.suppress(Exception)` instead of `try`-`except`-`pass`

(SIM105)